File: app/src/main/java/com/devdream/cookall/core/mapper/RecipeMapper.java
package com.devdream.cookall.core.mapper;

import com.devdream.cookall.core.api.responses.RecipeListResponse;
import com.devdream.cookall.core.dto.RecipeDTO;

import java.util.ArrayList;
import java.util.List;

public abstract class RecipeMapper {

    public static List<RecipeDTO> recipeListResponseToRecipeListDTO(
            List<RecipeListResponse> recipesListResponse) {
        List<RecipeDTO> recipes = new ArrayList<RecipeDTO>();

        for (RecipeListResponse recipeListResponse : recipesListResponse) {
            RecipeDTO recipe = new RecipeDTO();

            recipe.id = recipeListResponse.getId();
            recipe.profileId = recipeListResponse.getProfileId();
            recipe.title = recipeListResponse.getTitle();
            recipe.description = recipeListResponse.getDescription();
            recipe.cookingTime = recipeListResponse.getCookingTime();
            recipe.calories = recipeListResponse.getCalories();
            recipe.likes = recipeListResponse.getLikes();
            recipe.favorites = recipeListResponse.getFavorites();

            recipes.add(recipe);
        }
        return recipes;
    }

}

File: app/src/main/java/com/devdream/cookall/core/mapper/RecipeMapper.kt
package com.devdream.cookall.core.mapper

import com.devdream.cookall.core.api.responses.RecipeListResponse
import com.devdream.cookall.core.dto.RecipeDTO

import java.util.ArrayList

object RecipeMapper {

    fun recipeListResponseToRecipeListDTO(
            recipesListResponse: List<RecipeListResponse>): List<RecipeDTO> {
        val recipes = ArrayList<RecipeDTO>()

        for (recipeListResponse in recipesListResponse) {
            val recipe = RecipeDTO()

            recipe.id = recipeListResponse.id!!
            recipe.profileId = recipeListResponse.profileId!!
            recipe.title = recipeListResponse.title
            recipe.description = recipeListResponse.description
            recipe.cookingTime = recipeListResponse.cookingTime
            recipe.calories = recipeListResponse.calories!!
            recipe.likes = recipeListResponse.likes!!
            recipe.favorites = recipeListResponse.favorites!!

            recipes.add(recipe)
        }
        return recipes
    }

}
